pkg/internal/parser.go

```diff
@@ -162,37 +152,280 @@ func sanitize(groupName string) string {
+// api contains info about each type
+type api struct {
+	Name                      string
+	PkgName                   string
+	Version                   string
+	IsNamespaced              bool
+	AdditionalMethods         []AdditionalMethod
+	SkipVerbs                 []string
+	NoVerbs                   bool
+	HasStatus                 bool
+	ApplyConfigurationPackage string
+	writer                    io.Writer
+	importList                *[]string
+
+	PkgNameUpperFirst string
+	VersionUpperFirst string
+	NameLowerFirst    string
+	InputType         string
+	InputName         string
+	ResultType        string
+	Method            string
+}
+
+type AdditionalMethod struct {
+	Method      *string
+	Verb        *string
+	Subresource *string
+	Input       *string
+	Result      *string
+}
+
+func NewAPI(
+	root *loader.Package,
+	info *markers.TypeInfo,
+	group, version, applyconfigurationpkg string,
+	namespaceScoped bool,
+	additionalMethods []AdditionalMethod,
+	importList *[]string,
+	skipVerbs, onlyVerbs []string,
+	noVerbs, hasStatus bool,
+	w io.Writer,
+) (*api, error) {
 	typeInfo := root.TypesInfo.TypeOf(info.RawSpec.Name)
 	if typeInfo == types.Typ[types.Invalid] {
 		return nil, fmt.Errorf("unknown type: %s", info.Name)
 	}
 
 	api := &api{
-		Name:         info.RawSpec.Name.Name,
-		Version:      version,
-		PkgName:      group,
-		writer:       w,
-		IsNamespaced: isNamespaced,
-		HasStatus:    hasStatus,
+		Name:                      info.RawSpec.Name.Name,
+		PkgName:                   group,
+		Version:                   version,
+		ApplyConfigurationPackage: applyconfigurationpkg,
+		IsNamespaced:              namespaceScoped,
+		AdditionalMethods:         additionalMethods,
+		importList:                importList,
+		SkipVerbs:                 skipVerbs,
+		NoVerbs:                   noVerbs,
+		HasStatus:                 hasStatus,
+		writer:                    w,
 	}
 
 	api.setCased()
 	return api, nil
 }
 
+// TODO: add templating logic for additional methods
 func (a *api) WriteContent() error {
-	templ, err := template.New("wrapper").Parse(wrapperMethodsTempl)
+	generateApply := len(a.ApplyConfigurationPackage) > 0
+	// Add common template followed by specific methods which are required.
+	if err := templateExecute(wrapperMethodsTempl, *a); err != nil {
+		return err
+	}
+
+	if a.NoVerbs {
+		return nil
+	}
+
+	if a.hasVerb("get") {
+		if err := templateExecute(getTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("list") {
+		if err := templateExecute(listTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("watch") {
+		if err := templateExecute(watchTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("create") {
+		if err := templateExecute(createTemplate, *a); err != nil {
+			return err
+		}
+	}
+
+	if a.hasVerb("update") {
+		if err := templateExecute(updateTemplate, *a); err != nil {
+			return err
+		}
+	}
```

Comment:
Can't the conditionals for this live in the template itself?

Reply:
Explained here: https://github.com/kcp-dev/code-generator/pull/17#discussion_r883711128